src/main/java/io/liftbridge/Client.java

```diff
@@ -59,16 +83,68 @@ public void onCompleted() {}
     public void publish(String subject, byte[] payload,
                         long deadlineDuration, TimeUnit deadlineUnit) {
         Api.Message msg = Api.Message.newBuilder()
-            .setSubject(subject)
-            .setValue(ByteString.copyFrom(payload))
-            .setAckPolicy(Api.AckPolicy.LEADER)
-            .build();
+                .setSubject(subject)
+                .setValue(ByteString.copyFrom(payload))
+                .setAckPolicy(Api.AckPolicy.LEADER)
+                .build();
 
         Api.PublishRequest req = Api.PublishRequest.newBuilder()
-            .setMessage(msg)
-            .build();
+                .setMessage(msg)
+                .build();
 
         blockingStub.withDeadlineAfter(deadlineDuration, deadlineUnit)
-            .publish(req);
+                .publish(req);
+    }
+
+    /**
+     * {@code Builder} is used to configure and construct a {@link Client} instance.
+     */
+    public static class Builder {
```

Comment:
I think using a builder is a common idiom for configuring objects with a lot of options, which should be useful once the other client options are implemented.

Reply:
This also is meant to abstract away the gRPC implementation details from the user.

Reply:
It might be a good idea to have a builder for common options, but I wanted to leave the grpc channel option available to the user too since it gives a lot of flexibility (there's even examples of using consul's service discovery as the resolved for addresses). Also I think retries and stuff like that are kind of orthogonal to the client and could be accomplished by using other libraries on top of the basic client (like finagle-kafka does)

Reply:
Also adding options to the builder would mean a lot of new dependencies to the client (netty-grpc and some ssl library for example)

Reply:
Tracking on #3